File: Src/Icm.ContextConsole/DependencyResolver/IDependencyResolverExtensions.cs
using System;

public static class IDependencyResolverExtensions
{
	public static T GetService<T>(this IDependencyResolver resolver)
	{
		return (T)resolver.GetService(typeof(T));
	}
    
	public static T GetService<T>(this IDependencyResolver resolver, Type service)
	{
		return (T)resolver.GetService(service);
	}

}

File: Src2/Icm.ContextConsole/DependencyResolver/IDependencyResolverExtensions.vb
Imports System.Runtime.CompilerServices

Public Module IDependencyResolverExtensions

    <Extension>
    Function GetService(Of T)(resolver As IDependencyResolver) As T
        Return DirectCast(resolver.GetService(GetType(T)), T)
    End Function

    <Extension>
    Function GetService(Of T)(resolver As IDependencyResolver, service As Type) As T
        Return DirectCast(resolver.GetService(service), T)
    End Function

End Module
